metadata/src/main/java/org/apache/kafka/metadata/PartitionRegistration.java

```diff
@@ -310,11 +337,22 @@ public ApiMessageAndVersion toRecord(Uuid topicId, int partitionId, short versio
             setLeaderRecoveryState(leaderRecoveryState.value()).
             setLeaderEpoch(leaderEpoch).
             setPartitionEpoch(partitionEpoch);
-        if (version > 0) {
+        if (options.metadataVersion().isElrSupported()) {
             record.setEligibleLeaderReplicas(Replicas.toList(elr)).
                 setLastKnownELR(Replicas.toList(lastKnownElr));
         }
-        return new ApiMessageAndVersion(record, version);
+        if (options.metadataVersion().isDirectoryAssignmentSupported()) {
+            record.setDirectories(DirectoryId.toList(directories));
+        } else {
+            for (int i = 0; i < directories.length; i++) {
```

Comment:
Wondering if this is a `NullPointerException` waiting to happen since the current version of the PR allows `directories` to be null in some case as mentioned above?

Reply:
You're right. I've updated the constructor so that `directories` can never be null.